src/coreclr/gc/gc.cpp

```diff
@@ -20340,20 +20285,17 @@ bool gc_heap::try_get_new_free_region()
 bool gc_heap::init_table_for_region (int gen_number, heap_segment* region)
 {
 #ifdef BACKGROUND_GC
-        if (is_bgc_in_progress())
+        dprintf (GC_TABLE_LOG, ("new seg %Ix, mark_array is %Ix",
+            heap_segment_mem (region), mark_array));
+        if (((region->flags & heap_segment_flags_ma_committed) == 0) &&
+            !commit_mark_array_new_seg (__this, region))
```

Comment:
Shall we guard this under `gc_can_use_concurrent`?

Reply:
`commit_mark_array_new_seg` shouldn't do anything if concurrent is disabled; although I'm looking at the code, we are not initializing `background_saved_highest_address` in `init_gc_heap` which we should. @PeterSolMS could you please add this (ie, init `background_saved_highest_address` and `background_saved_lowest_address` to 0 in `init_gc_heap` like we do with WKS GC)? we could still guard this with `gc_can_use_concurrent` just to indicate our intention but it's not essential.

Reply:
I just pushed changes with the initialization of `background_saved_lowest_address` and `background_saved_highest_address`, and the restoration of  `current_total_committed_bookkeeping`.